tests/unit/utils/test_mssql_log_utils.py

```diff
@@ -40,10 +40,14 @@ def test_parse_statistics_messages():
             b" Scan count 1, logical reads 7, physical reads 0, read-ahead reads 0,"
             b" lob logical reads 0, lob physical reads 0, lob read-ahead reads 0."
         ),
+        # Newer version of SQL Server return IO stats in a different order and with
+        # additional items, so check we handle those correctly
         (
             b"Table '#tmp_1______________________________________________________________________________________________________________0000000014D9'."
-            b" Scan count 1, logical reads 4, physical reads 0, read-ahead reads 0,"
-            b" lob logical reads 0, lob physical reads 0, lob read-ahead reads 0."
+            b" Scan count 1, logical reads 4, physical reads 0, page server reads 0,"
```

Comment:
The way I've got it working at the moment is that it discards any stats it isn't expecting, because it only iterates the keys already in `cumulative_stats`:
https://github.com/opensafely-core/ehrql/blob/3995a653faecf706042bbe64cc0d6e3561cc1963/ehrql/utils/mssql_log_utils.py#L92-L101
https://github.com/opensafely-core/ehrql/blob/3995a653faecf706042bbe64cc0d6e3561cc1963/ehrql/utils/mssql_log_utils.py#L122-L123

I wanted to try to keep the output consistent between dev and prod in this respect.

I agree the tests here aren't as comprehensible as they could be, but they do cover everything and I didn't think it was worth the time making them better for a fairly small feature.